libs/core/src/main/java/org/opensearch/core/common/io/stream/ProtobufWriteable.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.core.common.io.stream;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+/**
+ * Implementers can be written to a {@linkplain OutputStream} and read from a {@linkplain byte[]}. This allows them to be "thrown
+ * across the wire" using OpenSearch's internal protocol with protobuf serialization/de-serialization.
+ *
+ * @opensearch.internal
+ */
+public interface ProtobufWriteable {
```

Comment:
Can we have this writeable extend `Writeable` interface?

Reply:
I agree with @Bukhtawar. These changes should be least intrusive and should use existing interfaces. It would enable us to experiment with other protocols as well.